src/iexamine.cpp

```diff
@@ -1711,20 +1711,25 @@ void iexamine::door_peephole( player &p, const tripoint &examp )
         return;
     }
 
-    // Peek through the peephole, or open the door.
-    const int choice = uilist( _( "Do what with the door?" ), {
-        _( "Peek through peephole." ),
-        _( "Open door." )
-    } );
-    if( choice == 0 ) {
-        // Peek
+    if( here.ter( examp )->open.is_null() && here.furn( examp )->open.is_null() ) {
```

Comment:
It would be better to use `map::open_door` with the `check_only` parameter set to `true`
```suggestion
    if( here.open_door( examp, true, true ) ) {
```
Better to not redefine the relevant logic when the function already allows for a dry run